src/lookup.py

```diff
@@ -687,162 +700,362 @@ def matches(self, entitytype=None, variant=None, subtype=None):
-                            or (
-                                validMissingSubtype
-                                and (
-                                    foundNameClean in givenNameClean
-                                    or givenNameClean in foundNameClean
-                                )
-                            )
-                        ):
-                            printf(
-                                "Loading entity, found name mismatch! In entities2: ",
-                                foundName,
-                                "; In BR: ",
-                                givenName,
-                            )
+                    ):
+                        printf(
+                            "Loading entity, found name mismatch! In entities2: ",
+                            foundName,
+                            "; In BR: ",
+                            givenName,
+                        )
+
+        overwrite = node.get("Overwrite") == "1"
 
-            entityConfig = self.lookupOne(entityType, variant, subtype)
-            if entityConfig:
+        entityConfig = self.lookupOne(entityType, variant, subtype)
+        if entityConfig:
+            if not overwrite:
                 printf(
                     f'Entity "{name}" from "{modName}" ({entityType}.{variant}.{subtype}) is overriding "{entityConfig.name}" from "{entityConfig.mod}"!'
                 )
-                entityConfig.fillFromXML(entity, resourcePath, modName, entityXML)
-            else:
-                entityConfig = self.EntityConfig(
-                    entity, resourcePath, modName, entityXML
+
+            entityConfig.fillFromXML(node, resourcePath, modName, entityXML)
+        else:
+            if overwrite:
+                printf(
+                    f'Entity "{name} from {modName} ({entityType}.{variant}.{subtype}) has the Overwrite attribute, but is not overwriting anything!'
                 )
 
-            return entityConfig
+            entityConfig = self.EntityConfig(node, resourcePath, modName, entityXML)
+            self.addEntity(entityConfig)
+
+        groups = []
+        nodeKind = node.get("Kind")
+        nodeGroup = node.get("Group")
+        if nodeKind is not None and nodeGroup is not None:
+            groups.append((nodeKind, nodeGroup))
+
+        for group in node.findall("group"):
+            kind = group.get("Kind", nodeKind)
+            name = group.get("Name", nodeGroup)
+            groups.append((kind, name))
+
+        for kind, group in groups:
+            groupName = f"({kind}) {group}"
+            tabGroupConfig = None
+            for tabGroup in self.tabGroups:
+                if tabGroup.isTab and tabGroup.name == kind:
+                    tabGroupConfig = tabGroup
+                    break
 
-        self.entityList.extend(list(map(mapEntity, entities)))
+            if tabGroupConfig is None:
+                tabGroupConfig = self.getGroup(name=kind, isTab=True)
```

Comment:
this naming conventions gets a bit confusing; tab makes sense, but there's two types of groups and kind is also a group?

Reply:
Yeah. Might just end up making tabs into their own class of things, they're similar to groups on paper but not so much in actual use.

Reply:
To elaborate, Kinds don't really exist in any meaningful way anymore. This section of the code is just back compatibility with them that converts them into tabs and fixes their group names to avoid conflicts.